samples/python/benchmark/bert_benhcmark/bert_benhcmark.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging as log
+from pathlib import Path
+import sys
+import tempfile
+from time import perf_counter
+
+import datasets
+from openvino.runtime import Core, get_version, AsyncInferQueue, PartialShape
+from transformers import AutoTokenizer
+from transformers.onnx import export
+from transformers.onnx.features import FeaturesManager
+
+
+def main():
+    log.basicConfig(format='[ %(levelname)s ] %(message)s', level=log.INFO, stream=sys.stdout)
+    log.info('OpenVINO:')
+    log.info(f"{'Build ':.<39} {get_version()}")
+    model_name = 'bert-base-uncased'
+    # Download the model
+    transformers_model = FeaturesManager.get_model_from_feature('default', model_name)
+    _, model_onnx_config = FeaturesManager.check_supported_model_or_raise(transformers_model, feature='default')
+    onnx_config = model_onnx_config(transformers_model.config)
+    # Download the tokenizer
+    tokenizer = AutoTokenizer.from_pretrained(model_name)
+
+    core = Core()
+
+    with tempfile.TemporaryDirectory() as tmp:
+        onnx_path = Path(tmp) / f'{model_name}.onnx'
+        # Export .onnx
+        export(tokenizer, transformers_model, onnx_config, onnx_config.default_onnx_opset, onnx_path)
+        # Read .onnx with OpenVINO
+        model = core.read_model(onnx_path)
+
+    # Enforce dynamic input shape
+    try:
+        model.reshape({model_input.any_name: PartialShape([1, '?']) for model_input in model.inputs})
+    except RuntimeError:
+        log.error("Can't set dynamic shape")
+        raise
+    # Optimize for throughput. Best throughput can be reached by
+    # running multiple openvino.runtime.InferRequest instances asyncronously
+    tput = {'PERFORMANCE_HINT': 'THROUGHPUT'}
+    # Pick a device by replacing CPU, for example MULTI:CPU(4),GPU(8).
+    # It is possible to set CUMULATIVE_THROUGHPUT as PERFORMANCE_HINT for AUTO device
+    compiled_model = core.compile_model(model, 'CPU', tput)
+    # AsyncInferQueue creates optimal number of InferRequest instances
+    ireqs = AsyncInferQueue(compiled_model)
+
+    sst2 = datasets.load_dataset('glue', 'sst2')
+    sst2_sentences = sst2['validation']['sentence']
+    # Warm up
+    encoded_warm_up = dict(tokenizer('Warm up sentence is here.', return_tensors='np'))
+    for _ in ireqs:
+        ireqs.start_async(encoded_warm_up)
+    ireqs.wait_all()
+    # Benchmark
+    start = perf_counter()
+    for sentence in sst2_sentences:
+        encoded = dict(tokenizer(sentence, return_tensors='np'))
+        ireqs.start_async(encoded)
+    ireqs.wait_all()
+    end = perf_counter()
+    log.info(f'Duration:       {end - start:.2f} seconds')
```

Comment:
Are you sure that results are valid if no callback is specified?
Items per-second should be calculated and reported as well, like "Throughput: 100 it/sec"

Reply:
> Are you sure that results are valid if no callback is specified?

I agree that the pipeline is incomplete. The full pipeline should include postprocessing. But that makes the sample more complicated and that's how benchmark_app works.

> Items per-second should be calculated and reported as well, like "Throughput: 100 it/sec"

It doesn't make much sense to report throughput because each sample is of different length. Instead I report how long it takes to process a given dataset

Reply:
At least average time and average sequence length would be helpful to see. I would also recommend looking at what other frameworks have in this case. Here is an example: https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/python/tools/transformers/benchmark.py

Reply:
Updated.

onnxruntime's benchmark is more complex. Maybe it will be useful to compare onnxruntime and our benchmark_app tool